library/common/event/provisional_dispatcher.cc

```diff
@@ -49,5 +49,21 @@ void ProvisionalDispatcher::deferredDelete(DeferredDeletablePtr&& to_delete) {
   event_dispatcher_->deferredDelete(std::move(to_delete));
 }
 
+void ProvisionalDispatcher::pushTrackedObject(const ScopeTrackedObject* object) {
+  if (event_dispatcher_) {
+    event_dispatcher_->pushTrackedObject(object);
+  }
+}
+
+void ProvisionalDispatcher::popTrackedObject(const ScopeTrackedObject* expected_object) {
+  if (event_dispatcher_) {
+    event_dispatcher_->popTrackedObject(expected_object);
+  }
+}
+
+bool ProvisionalDispatcher::trackedObjectStackIsEmpty() const {
```

Comment:
When does this get called?

Reply:
Based on your check below, we might want to lock here, depending on the usage.

Reply:
This actually never gets currently called. I could change usage to NOT_REACHED. But for now I changed to isThreadSafe assertion.